manta-pay/concrete-protocol.tex

```diff
@@ -4,46 +4,79 @@
-    where $x$ is the coordinate-extraction function from $\mathbb{G}$ to $\mathbb{F}$.
-    For the Void Number Commitment Scheme we use an arity-2 \Poseidon{} with the following mapping:
-    \[\COM^{\VN}(\sk, \cm) := \Poseidon_2(\lift(\sk), \cm)\]
-\end{definitiontoc}
+\begin{definition}[Commitment Schemes]
+    The protocol features two different commitment schemes: $\COM^{\UTXO}$ the \UTXO{} Commitment Scheme and $\COM^{\VN}$ the Void Number Commitment Scheme. Both commitment schemes use \Poseidon{} as the underlying cryptographic primitive. The \UTXO{} uses an arity-8 \Poseidon{} with the following mapping:
+    \[\COM^{\UTXO}_r(\textsf{D}, \pkD, \asset) := \Poseidon_8(d, 0, r, \X(\textsf{D}), \Y(\textsf{D}), \X(\pkD), \Y(\pkD), \asset.\ID, \asset.\VALUE)\]
+    where $d = \HashToScalar(\domainstr{manta-pay/1.0.0/com-utxo})$.
+    For the Void Number Commitment Scheme we use an arity-4 \Poseidon{} with the following mapping:
+    \[\COM^{\VN}_\ak(\cm) := \Poseidon_4(\HashToScalar(\domainstr{manta-pay/1.0.0/com-vn}), 0, \X(\ak), \Y(\ak), \cm)\]
+\end{definition}
+
+\begin{definition}[Key-Derivation Functions]
+    For the encryption scheme \KDF{s}, we use the following which maps a group element $G : \mathbb{G}$ to a scalar:
+    \[\KDF(G) := \Poseidon_2(\HashToScalar(\domainstr{manta-pay/1.0.0/encryption-kdf}), x(G), y(G))\]
+\end{definition}
+
+\begin{definition}[Randomizable Key-Derivation Function]
+    TODO: Brandon
+\end{definition}
 
-\begin{definitiontoc}{Key-Agreement Scheme}
+\begin{definition}[Key-Agreement Scheme]
     For $\KA$, we use a Diffie-Hellman Key Exchange over $(\mathbb{G}, \mathbb{S})$:
     \begin{align*}
         \KA.\derive(x)   &: \mathbb{S} \to \mathbb{G}                   := x \cdot G \\
         \KA.\agree(x, Y) &: \mathbb{S} \times \mathbb{G} \to \mathbb{G} := x \cdot Y
     \end{align*}
     where $G$ is a fixed public point.
-\end{definitiontoc}
+\end{definition}
 
-\begin{definitiontoc}{Symmetric-Key Encryption Scheme}
-    For $\SYM$, we use symmetric-key encryption scheme: AES-GCM~\cite{rfc5288} with magic-number nonce and no associated data. Note that it is safe to reuse the nonce here because we assume that the encryption key is only used once. See \autoref{def:abstract-symm} for more.
-\end{definitiontoc}
+\begin{definition}[Message Authentication Code]
+    For message authentication codes we use the following instantiation of \Poseidon{}:
+    \[\MAC(\sk, m) := \Poseidon_{\abs{m} + 1}(\HashToScalar(\domainstr{manta-pay/1.0.0/mac}), \sk, m)\]
 
-\begin{definitiontoc}{Key-Derivation Functions}
-    For $\KDF$, we use Blake2s~\cite{rfc7693} with magic-number salt.
-\end{definitiontoc}
+    In this protocol, we use $|m| \in \set{2, 6}$ for $\OutgoingNote$ and $\IncomingNote$ respectively.
+\end{definition}
 
-\begin{definitiontoc}{Dynamic Cryptographic Accumulator}
+\begin{definition}[Signature Scheme]
+    For the signature scheme we use standard ECDSA over $\mathbb{G}$.
```

Comment:
We are using Schnorr signature instead of ECDSA?

Reply:
Schnorr is just a lot better: https://bitcoin.stackexchange.com/questions/77234/what-are-the-advantages-of-schnorr-vs-ecdsa